crates/diesel_models/src/query/role.rs

```diff
@@ -115,9 +138,10 @@ impl Role {
 
         if let Some(merchant_id) = merchant_id {
             query = query.filter(
-                dsl::merchant_id
+                (dsl::merchant_id
                     .eq(merchant_id)
-                    .or(dsl::scope.eq(RoleScope::Organization)),
+                    .and(dsl::scope.eq(RoleScope::Merchant)))
+                .or(dsl::scope.eq(RoleScope::Organization)),
```

Comment:
Please test this query.

Reply:
Checked , it gives the expected response